### PR TITLE
tests(fuzz): Add `parse_duration` fuzzing target

### PR DESCRIPTION
**Pull request by caspermeijn**
Add a fuzzing target for parsing into a Duration.

### CODE REVIEW

**Comment by caspermeijn** [comment]
@mumbleskates This is the other fuzzer I used for my testing. Do you have suggestions?

**Review by mumbleskates — COMMENTED** [review]
this looks pretty good. in the fuzzer i have committed so far i also added some regex that helps assert that anything we successfully parsed actually looks reasonable and like we expect, so we don't end up accidentally parsing something that looks really strange. for this case that should be a super simple pattern.

**Comment by caspermeijn** [comment]
> this looks pretty good. in the fuzzer i have committed so far i also added some regex that helps assert that anything we successfully parsed actually looks reasonable and like we expect, so we don't end up accidentally parsing something that looks really strange. for this case that should be a super simple pattern.

I found that the regex made the test less readable. If we can roundtrip the text, then it can't be that strange, right?

File: fuzz/fuzzers/parse_duration.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+pub fn test_parse_duration(data: &[u8]) {
+    use std::str::from_utf8;
+    use std::str::FromStr;
+
+    // input must be text
+    let Ok(original_text) = from_utf8(data) else {
+        return;
+    };
+
+    // parse input as a datetime
```
[the remark this answers is not in the document]
Comment (caspermeijn) [reply fuzz/fuzzers/parse_duration.rs]: Thanks!

**Comment by mumbleskates** [comment]
that is a reasonable enough logic! i find regex pretty tractable personally but that obviously varies wildly from person to person. works great without it :+1:

**Pull request merged**